src/pages/ShoppingCart.page.js

```diff
@@ -0,0 +1,42 @@
+import { InventoryPage } from './Inventory.page';
+
+export class ShoppingCartPage extends InventoryPage {
+    url = '/cart.html';
+
+    cartItemSelector = '.cart_item';
+
+    removeItemSelector = '[id^="remove"]';
+
+    get headerTitle() { return this.page.locator('.title'); }
+
+    get cartItems() { return this.page.locator('.cart_item'); }
+
+    get checkoutButton() { return this.page.locator('#checkout'); }
+
+    // async below added to show the function returns a promise
+    async getCartItemByName(name) { return this.page.locator(this.cartItemSelector, { hasText: name }); }
+
+    async removeCartItemByName(name) {
+        const item = await this.getCartItemByName(name);
+        return item.locator(this.removeItemSelector);
+    }
+
+    async removeCartItemById(id) {
+        await this.cartItems.nth(id).locator(this.removeItemSelector).click();
+    }
+
+    async getCartItemsList() {
+        const items = await this.cartItems.all();
+        return Promise.all(items.map(async (item) => {
```

Comment:
Right now you have 3 similar methods, let's think about the way to combine them (maybe not all of them :) ) into one and reuse that code